src/main/kotlin/model/Bet.kt

```diff
@@ -0,0 +1,15 @@
+package model
+
+@JvmInline
+value class Bet(val amount: Double) {
+    init {
+        require(amount >= 0) { "Amount should be greater than 0" }
+    }
+
+    companion object {
+        fun parseToBet(input: String): Bet {
+            val amount = input.trim().toDouble()
+            return Bet(amount)
+        }
+    }
+}
```

Comment:
> Kotlin's value class constructors can't be made private, so while we can enforce validation in the init block, we can't fully prevent misuse unless we wrap creation in a factory method. It's a trade-off between convenience and safety. What would you suggest?

I think the View layer can handle null or blank inputs and convert them to `Double`.
Just like -- how browsers send a `Boolean` instead of a string when toggling switch button.